Source/WebCore/testing/Internals.cpp

```diff
@@ -2924,11 +2928,12 @@ void Internals::setAutomaticLinkDetectionEnabled(bool enabled)
 #endif
 }
 
-bool Internals::testProcessIncomingSyncMessagesWhenWaitingForSyncReply()
+ExceptionOr<bool> Internals::testProcessIncomingSyncMessagesWhenWaitingForSyncReply()
 {
-    ASSERT(contextDocument());
-    ASSERT(contextDocument()->page());
-    return contextDocument()->page()->chrome().client().testProcessIncomingSyncMessagesWhenWaitingForSyncReply();
+    RefPtr document = contextDocument();
+    if (!document || !document->page())
+        return Exception { ExceptionCode::InvalidStateError };
```

Comment:
Is it important that this throw an exception rather than simply returning `false`?

Reply:
I'm not sure and don't have strong opinion on this. Currently, if we can't get the doc or page this would crash in debug mode so webkit developers can catch the issue. My idea was that throwing an exception would keep a similar hint that something bad is happening. If we just return false, probably it's harder to understand when such an issue is happening.

Reply:
Crashes often come with a backtrace; exceptions do not do all that much to help someone identify what went wrong in a test. I suppose the main argument in favor of an exception is that just returning false might cause a false negative where a test would pass that should fail.